python_modules/dagster-graphql/dagster_graphql/implementation/fetch_sensors.py

```diff
@@ -190,28 +184,19 @@ def get_sensor_next_tick(
     from dagster_graphql.schema.instigation import GrapheneDryRunInstigationTick
 
     check.inst_param(sensor_state, "sensor_state", InstigatorState)
-
-    repository_origin = sensor_state.origin.repository_origin
-    if not graphene_info.context.has_code_location(
-        repository_origin.code_location_origin.location_name
-    ):
+    if not sensor_state.is_running:
         return None
 
-    code_location = graphene_info.context.get_code_location(
-        repository_origin.code_location_origin.location_name
+    repository_origin = sensor_state.origin.repository_origin
+    selector = SensorSelector(
+        location_name=repository_origin.code_location_origin.location_name,
+        repository_name=repository_origin.repository_name,
+        sensor_name=sensor_state.name,
     )
-    if not code_location.has_repository(repository_origin.repository_name):
-        return None
-
-    repository = code_location.get_repository(repository_origin.repository_name)
 
-    if not repository.has_sensor(sensor_state.name):
-        return None
-
-    sensor = repository.get_sensor(sensor_state.name)
-
-    if not sensor_state.is_running:
+    if not graphene_info.context.has_sensor(selector):
         return None
+    sensor = graphene_info.context.get_sensor(selector)
```

Comment:
this came up on the other PR but this has then get pattern is a little weird (esp if they both interact with a cache potentially), why not get and then handle it being None and make it more atomic?

Reply:
good call, will change